cmd/otelcol/config/collector/agent_config.yaml

```
@@ -71,7 +71,7 @@ receivers:
       - job_name: 'otel-collector'
         scrape_interval: 10s
         static_configs:
-        - targets: ["${SPLUNK_LISTEN_INTERFACE}:8888"]
+        - targets: ["0.0.0.0:8888"]
```

```suggestion
        - targets: ["localhost:8888"]
```
I know 0.0.0.0 will route to localhost but using localhost explicitly kind of conveys that the listener is probably only bound to it. It's just a minor thing, so feel free to merge as-is.

---

Yeah, I was thinking about it as well. Two reasons I decided against it:
1. It'd require updating all other configs for consistency (it's a lot of them)
2. I added a comment on how to expose the metrics under a different endpoint if needed. `0.0.0.0` would work with any interface they chose. Otherwise, they would need to change this part as well.

---

I'm actually not 100% sure about point 2, targeting `0.0.0.0` may not work with specific IP addresses, only with `localhost` and `0.0.0.0`, which is no different from targeting `localhost`. If that's the case, we can update all the configs from targeting `0.0.0.0` to `localhost` separately.

---

I am mostly certain 0.0.0.0 as destination address only works if server is listening on localhost